.atomist/editors/ConvertManifestToPackageJson.ts

```diff
@@ -0,0 +1,103 @@
+        if (project.fileExists(manifestYamlName)) {
+            const manifestContent = project.findFile(manifestYamlName).content;
+            const manifestYaml = yaml.load(manifestContent);
+            const packageJson = this.loadPackageJson(project);
+
+            const newPackageJson: any = {};
+            // First we add name and version to the top of the package.json
+            newPackageJson.name = `@${manifestYaml.group}/${manifestYaml.artifact}`;
+            newPackageJson.version = manifestYaml.version;
+
+            // Then we copy all existing keys into the new package.json
+            // tslint:disable-next-line:forin
+            for (const key in packageJson) {
+                newPackageJson[key] = packageJson[key];
+            }
+
+            // Create or retrieve atomist section from package.json
+            let atomist: any = {};
+            if (newPackageJson.atomist == null) {
+                newPackageJson.atomist = atomist;
+            } else {
+                atomist = newPackageJson.atomist;
+            }
+
+            atomist.requires = manifestYaml.requires;
+
+            // Add in dependencies
+            if (manifestYaml.dependencies != null) {
+                atomist.dependencies = {};
+                for (const d of manifestYaml.dependencies) {
+                    const parts = (d as string).split(":");
+                    atomist.dependencies[`${parts[0]}:${parts[1]}`] = parts[2];
+                }
+            }
+
+            // Add in extensions
+            if (manifestYaml.extensions != null) {
+                atomist.extensions = {};
+                for (const d of manifestYaml.extensions) {
+                    const parts = (d as string).split(":");
+                    atomist.extensions[`${parts[0]}:${parts[1]}`] = parts[2];
+                }
+            }
+
+            if (manifestYaml.excludes != null) {
+                atomist.excludes = {};
+                // The json keys don't use _. So we convert them to -
+                atomist.excludes.editors = manifestYaml.excludes.editors;
+                atomist.excludes.generators = manifestYaml.excludes.generators;
+                atomist.excludes["command-handlers"] = manifestYaml.excludes.command_handlers;
+                atomist.excludes["event-handlers"] = manifestYaml.excludes.event_handlers;
+                atomist.excludes["response-handlers"] = manifestYaml.excludes.response_handlers;
+            }
+
+            const newPackageJsonContent = JSON.stringify(newPackageJson, null, 2);
+            project.findFile(packageJsonName).setContent(newPackageJsonContent);
+            project.deleteFile(manifestYamlName);
+        }
+    }
+
+    private loadPackageJson(project: Project): any {
+        if (project.fileExists(packageJsonName)) {
+            return JSON.parse(project.findFile(packageJsonName).content);
+        } else {
+            project.addFile(packageJsonName, "{}");
```

Comment:
Might be better to copy the one from this archive to give a more reasonable starting point.